.circleci/config.yml

```diff
@@ -26,7 +26,9 @@ commands:
       - run: go build ./...
   go_test:
     steps:
-      - run: go test -v ./...
+      - run:
+          no_output_timeout: 15m
+          command: go test -timeout=15m -v ./...
```

Comment:
The `go test` timeout increase here is needed because it's a timeout for the whole package, not for a single test and the time it takes to run all tests within `e2etest` has (naturally) increased with another version in the mix. The default timeout is 10m and we would regularly reach ~8m without 0.15.

A cleaner solution here would be:
1. decoupling E2E tests entirely and make sure they never run by default on `go test` (without additional filtration)
2. create a step which runs just quick unit tests and that step can have default 10m timeout, or even lower
3. create a step which runs just E2E tests - here we can bump the timeout to whatever necessary

A followup on (3) could be also running these tests in parallel somehow in CircleCI, which could possibly help us bypass the per-package timeout? I'm not sure about that though - I don't have any experience with this.

Reply:
I think this solution is fine, but have created https://github.com/hashicorp/terraform-exec/issues/145 in case we want to investigate (3).